app-admin/super-user-spark: Add 0.4.0.1

hololeap opened the pull request:
Upstream finally bumped the version to be compatible with GHC 8.4.3

This will hopefully resolve #823

solpeth commented:
Could you take a look at the CI failures please? Looks like some missing dependencies.

Running `repoman commit` to commit your changes will check for this before allowing you to commit.

hololeap commented:
I guess I missed a dependency. Sorry!

It's passing now. I will use `repoman commit` in the future.

solpeth commented:
@hololeap it merges successfully, which is great! However, with `FEATURES=test` enabled I'm running into some problems.

There's a circular dependency issue with `genvalidity-vector` depending on itself when `test` is enabled, so the test deps will need to be commented out and `test-suite` removed from the `CABAL_FEATURES` line at the top of that ebuild.

`super-user-spark` is also trying to downgrade `hspec` and `hspec-core` to 2.4.8, when the overlay is currently on 2.5.0. Could you please try and figure out if `super-user-spark`'s test-suite will run on `hspec-2.5`?

I haven't come across anything else so far. Thanks!

hololeap commented:
Ok. I made all the changes you requested. I dropped the dependency of `genvalidity-vector` on itself and I changed the deps of `super-user-spark` to allow for `hspec-2.5.0`. Everything builds successfully with `USE=test FEATURES=test` on my system.

solpeth commented:
Great work! Pulled.

Thanks!
